### PR TITLE
Fix #905: correctly manage the output encoding for external commands

### LINKED ISSUE #905: External commands passive checks encoding
When Alignak receives an external command for a passive service check that contains some special characters in the reported plugin output, the receiver daemon raises an exception when the command is submitted to the scheduler (as reported in #887).

No exception is raised when the monitoring logs are disabled for the passive checks in the `alignak.cfg` file.

This is mainly due to an incorrect management of the plugin output encoding that is of no importance if monitoring logs are not activated.

### PR DESCRIPTION
**Pull request by mohierf**
As reported by @spea1 ... special characters in passive plugin output may break the Alignak receiver daemon.

This pull request improves the external commands plugin output decoding and adds some tests for this problem

### CODE REVIEW

**Comment by mohierf** [comment]
Expected behavior is confirmed by @spea1

File: alignak/external_command.py [hunk 1]
```diff
@@ -2973,49 +2973,61 @@ def process_host_check_result(self, host, status_code, plugin_output):
...
+        # If globally disabled OR host disabled, do not launch..
+        if not cls.accept_passive_checks or not host.passive_checks_enabled:
+            return
+
+        try:
+            plugin_output = plugin_output.decode('utf8', 'ignore')
+            logger.debug('%s > Passive host check plugin output: %s',
+                         host.get_full_name(), plugin_output)
+        except UnicodeError:
+            pass
```
Comment (ddurieux) [inline alignak/external_command.py]: It can be nice to have a warning / error

File: alignak/external_command.py [hunk 2]
```diff
@@ -3044,51 +3056,69 @@ def process_service_check_result(self, service, return_code, plugin_output):
...
+            logger.debug('%s > Passive service check plugin output: %s',
+                         service.get_full_name(), plugin_output)
+        except UnicodeError:
+            pass
+
+        # Maybe the check is just too old, if so, bail out!
+        if self.current_timestamp < service.last_chk:
+            logger.warning('%s > Passive service check is too old. Ignoring...',
+                           service.get_full_name())
+            return
```
Comment (ddurieux) [inline alignak/external_command.py]: same here

File: alignak/external_command.py (hunk 1 above)
Comment (mohierf) [reply alignak/external_command.py]: It is of no interest... except for spamming the log 😉

**Pull request merged**